app/controllers/admin/tags_controller.rb

```diff
@@ -18,10 +19,26 @@ def index
       @tags = @tags.where("tags.name ILIKE :search", search: "%#{params[:search]}%") if params[:search].present?
     end
 
+    def new
+      @tag = Tag.new
+    end
+
+    def create
+      @tag = Tag.new(tag_params)
+      @tag.name = params[:tag][:name].downcase
+
+      if @tag.save
+        flash[:success] = "Tag has been created!"
+        redirect_to admin_tag_path(@tag)
+      else
+        flash[:danger] = @tag.errors_as_sentence
+        render :new
+      end
+    end
+
     def show
```

Comment:
now that `show` has a form, I wonder if we should just switch it to `edit`, as many of other admin pages only have the edit page. `show` should be to present read only data

Reply:
I'm down for making this change and separating these 👍

Reply:
I think just having `/edit` instead of show would be enough. Most admin pages don't have a show view, only the edit. No need to add it IMO